src/modules/endpoint/MemoryDatabase.js

```diff
@@ -45,12 +48,55 @@ class MemoryDatabase {
     return this._categoriesMap
   }
 
+  set categoriesMap (categoriesMap: CategoriesMapModel) {
+    if (this._categoriesMap !== null) {
+      throw Error('categoriesMap has already been set on this context!')
+    }
+    this._categoriesMap = categoriesMap
+  }
+
   get languages (): Array<LanguageModel> {
     return this._languages
   }
 
-  get resourceCache (): ResourceCacheType {
-    return this._resourceCache
+  set languages (languages: Array<LanguageModel>) {
+    if (this._languages !== null) {
+      throw Error('languages has already been set on this context!')
+    }
+    this._languages = languages
+  }
+
+  get categoriesResourceCache (): ResourceCacheType {
+    return this._categoriesResourceCache
+  }
+
+  set categoriesResourceCache (resourceCache: ResourceCacheType) {
+    if (this._categoriesResourceCache !== null) {
+      throw Error('categoriesResourceCache has already been set on this context!')
+    }
+    this._categoriesResourceCache = resourceCache
+  }
+
+  get events (): Array<EventModel> {
+    return this._events
+  }
+
+  set events (events: Array<EventModel>) {
+    if (this._events !== null) {
+      throw Error('events has already been set on this context!')
+    }
+    this._events = events
+  }
+
+  get eventsResourceCache (): ResourceCacheType {
```

Comment:
We'll probably have only one resource cache. I'll fix this in my current task though.